### DISCUSSION TITLE
create a regression model for the concrete data using keras

### DISCUSSION THREAD

**Discussion by Madeha9** [keras-team/keras#20199] [Q&A] [answered]
Hello everyone,,
I'm  building my first regression model using keras , and my dataset is the concrete dataset , everything was working perfectly, I excute the code many times by manipulating the number of the hidden layers and the epochs numbers , after a while I started getting an Attribute error message with fit function can you help me please to point where is the error exactly , attached is the code 
![Screenshot 2024-09-03 at 10 39 58](https://github.com/user-attachments/assets/42506701-78c6-45ab-86da-2bd43d2ee751)
![Screenshot 2024-09-03 at 10 39 58](https://github.com/user-attachments/assets/e14e143d-4621-46ab-a437-35e72fa6558a)

**Comment by santi-mir** [comment]
`model.fit` fails because the `model` variable does not store the Model class, but instead it's NoneType class.

We would need to see the full code, or at least were you assign `model` variable, to debug it

**Reply by Madeha9** [reply]
[regresssion project_D .pdf](https://github.com/user-attachments/files/16868542/regresssion.project_D.pdf)
Hey, thanks for your response , attached is the full code , I hope that someone could help I have being trying for almost a week and I'm not able to figure it out 
 
![Screenshot 2024-09-04 at 12 14 26](https://github.com/user-attachments/assets/e1ea896a-34b2-4141-93e1-2200c8baa96a)
![Screenshot 2024-09-04 at 12 15 24](https://github.com/user-attachments/assets/76c51ad8-0bcd-4841-a522-a089b493da0a)
![Screenshot 2024-09-04 at 12 16 22](https://github.com/user-attachments/assets/768ea01d-ce07-479e-9849-8be307ee57e4)

**Reply by muriloasouza** [reply, marked as answer]
Aren't you missing `return model` in your `regression_model()` function?

**Reply by Madeha9** [reply]
yes you right, I did not add the rentrun model  and that was issue , I added  the return , the code runs normally .

thank you so much appreciate it

**Reply by Madeha9** [reply]
> `model.fit` fails because the `model` variable does not store the Model class, but instead it's NoneType class.
> 
> We would need to see the full code, or at least were you assign `model` variable, to debug it

thank you , I was misising the return  step
